privaterelay/management/commands/update_phone_remaining_stats.py

```diff
@@ -3,58 +3,88 @@
+    get_phone_subscriber_social_accounts,
+)
 
 if settings.PHONES_ENABLED:
     from phones.models import RelayNumber
 
-from privaterelay.views import update_fxa
+logger = logging.getLogger("events")
+
+
+def reset_phone_remaining_stats(user):
+    # re-set remaining_texts and remaining_seconds to the maximum value
+    try:
+        relay_number = RelayNumber.objects.get(user=user)
+    except RelayNumber.DoesNotExist:
+        # no RelayNumber set, do nothing
+        return
+    relay_number.remaining_texts = settings.MAX_TEXTS_PER_BILLING_CYCLE
+    relay_number.remaining_seconds = settings.MAX_MINUTES_PER_BILLING_CYCLE * 60
+    relay_number.save()
 
 
-def update_phone_remaining_stats():
-    if not settings.PHONES_ENABLED:
-        return []
-    accounts_with_phones = []
-    for sub_with_phone in settings.SUBSCRIPTIONS_WITH_PHONE:
-        social_accounts = SocialAccount.objects.filter(
-            extra_data__icontains=sub_with_phone
+def get_next_reset_date(profile: Profile) -> datetime:
+    # TODO: consider moving this as a property in Profile model
+    # assumes that profile being passed have already been checked to have
+    # phone subscription or a free phone user
+    if profile.date_phone_subscription_reset is None:
+        # there is a problem with the sync_phone_related_dates_on_profile
+        # or a new foxfooder whose date_phone_subscription_reset did not get set in
+        logger.error(
+            "phone_user_profile_dates_not_set",
+            extra={
+                "fxa_uid": profile.fxa.uid,
+                "date_subscribed_phone": profile.date_phone_subscription_end,
+                "date_phone_subscription_start": profile.date_phone_subscription_start,
+                "date_phone_subscription_reset": profile.date_phone_subscription_reset,
+                "date_phone_subscription_end": profile.date_phone_subscription_end,
+            },
         )
-        accounts_with_phones.extend(list(social_accounts))
+        return datetime.now(timezone.utc) - timedelta(minutes=15)
+
+    calculated_next_reset_date = profile.date_phone_subscription_reset + timedelta(
+        settings.MAX_DAYS_IN_MONTH
+    )
+    if profile.date_phone_subscription_end is None:
+        return calculated_next_reset_date
+    if profile.date_phone_subscription_end < calculated_next_reset_date:
+        # return the past or the closest next reset date
+        return profile.date_phone_subscription_end
+    return calculated_next_reset_date
+
+
+def update_phone_remaining_stats() -> tuple[int, int]:
+    accounts_with_phones = get_phone_subscriber_social_accounts()
+    free_phones_social_accounts = get_free_phone_social_accounts()
+    accounts_with_phones.update(free_phones_social_accounts)
+
+    if not settings.PHONES_ENABLED or len(accounts_with_phones) == 0:
+        return 0, 0
+
     updated_profiles = []
+    datetime_now = datetime.now(timezone.utc)
     for social_account in accounts_with_phones:
```

Comment:
suggestion (non-blocking): If we had a`Profile.objects.need_phone_stats_update()` method, this would be a spot we could use it, instead of querying and looping over EVERY phone social account record?

Reply:
This is a good point. I am going to add this as a separate follow-up ticket to reconsider. I would like to optimize this after it's tested and found to be working. The flip side to this is we could record the next calculated reset date on the profile and grab optimize for that which would also help Frontend engineers use the field to show the next reset date, but because I was unsure how the dates will be used I thought it would be best to keep the dates simple and calculate future reset dates if needed (yearly and foxfeeders).

Reply:
Created a follow-up ticket in our tech backlog: https://mozilla-hub.atlassian.net/browse/MPP-2972